src/network/clientpackethandler.cpp

```diff
@@ -530,11 +531,21 @@ void Client::handleCommand_Movement(NetworkPacket* pkt)
 void Client::handleCommand_Fov(NetworkPacket *pkt)
 {
 	f32 fov;
-	bool is_multiplier;
+	bool is_multiplier = false;
+	f32 transition_time = 0.0f;
+
 	*pkt >> fov >> is_multiplier;
 
+	// Wrap transition_time extraction within a
+	// try-catch to preserve backwards compat
+	try {
+		*pkt >> transition_time;
+	} catch (PacketError &e) {};
```

Comment:
The try-catch here is actually redundant - if a set_fov call doens't explicitly specify `transition_time`, it is assumed to be `0.0f` before it is even sent over the network.

https://github.com/minetest/minetest/pull/9705/files#diff-9444313da8be4290e9304a4b1faed804R1267

My question now, is whether I should remove this try-catch, or prevent sending the 0 in `Server::sendPlayerFov`. The former is pretty easy, while the latter might require detecting the presence of `time` and setting the packet size accordingly.

Reply:
It's absolutely not redundant because a 5.2 server will not send a transition_time.

> require detecting the presence of time and setting the packet size accordingly

The value passed to the packet constructor is just a preallocation, nothing bad happens if you put more or less bytes into the packet.

Reply:
Oh yes, of course >.<  ...I still can't fully wrap my brain around the concept of backwards compat :)

> The value passed to the packet constructor is just a preallocation, nothing bad happens if you put more or less bytes into the packet.

Ah, nice. So, shall I remove the initialising-to-zero in `ObjectRef::l_set_fov` then? If the try-catch is here to stay, it's that part of the code that's redundant, I guess.

Reply:
Nah, the server should send a complete packet even if the last value is `0.0f`

Reply:
Ok, awesome.